test/contrib/test_module.py

```diff
@@ -0,0 +1,43 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import haiku as hk
+import flax
+from numpyro.contrib.module import haiku_module, flax_module
+import numpy as np
+from numpyro import handlers
+import numpyro
+
+
+X = np.array(list(range(100)))
+Y = np.array([x*2 + 2.0 for x in list(range(100))])
+param_keys = ['type', 'name', 'fn', 'args', 'kwargs', 'value', 'scale', 'cond_indep_stack']
+
+
+def haiku_model(x, y):
+    linear_module = hk.transform(lambda x: hk.Linear(1)(x))
+    nn = haiku_module("nn", linear_module, 1)
+    assert nn.args[0]
+    mean = nn(x)
+    numpyro.sample("y", numpyro.distributions.Normal(mean, 0.1), obs=y)
+
+
+def flax_model(x, y):
+    linear_module = flax.nn.Dense.partial(features=1)
+    nn = flax_module("nn", linear_module, 1)
+    mean = nn(x)
+    numpyro.sample("y", numpyro.distributions.Normal(mean, 0.1), obs=y)
+
+
+def test_module():
```

Comment:
Could you separating this into 2 different tests `test_flax_module` and `test_haiku_module`?

Reply:
separate functions within `test_module.py`, or two separate files ?

Reply:
I meant into 2 functions. We rarely combine tests unless doing so will improve testing time.